src/main/java/com/internet/shop/controllers/admin/DeleteUsersController.java

```diff
@@ -12,15 +12,21 @@
 
 public class DeleteUsersController extends HttpServlet {
     public static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static final String USER_ID = "user_id";
     private UserService userService = (UserService) injector.getInstance(UserService.class);
     private ShoppingCartService shoppingCartService
             = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        String userId = req.getParameter("id");
-        Long id = Long.valueOf(userId);
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        String userToDeleteId = req.getParameter("id");
+        Long id = Long.valueOf(userToDeleteId);
+        if (userId.equals(id)) {
+            req.getRequestDispatcher("/WEB-INF/views/accessDenied.jsp").forward(req, resp);
+            return;
+        }
         userService.delete(id);
         ShoppingCart cart = shoppingCartService.getByUserId(id);
         shoppingCartService.delete(cart.getId());
```

Comment:
let's first delete shopping cart and then user, cause shopping cart depends on user and the order will be important when we will start working with the real database